### PR TITLE
feat(color): add a foreground color option

### PR DESCRIPTION
**Pull request by redboul**

### CODE REVIEW

**Comment by stewartlord** [comment]
Thank you so much for the PR. I'm just traveling at the moment. I'll take a closer look shortly.

File: README.md [hunk 1]
```diff
@@ -37,6 +38,7 @@ To customize additional properties, generate the Identicon by supplying a hash s
 // set up options
 var hash = "myUnicodeUsername!";          // Any unicode string
 var options = {
+      foreground: [0, 0, 0, 255],  // rgba black
```
Comment (stewartlord) [inline README.md]: Nitpicking here, I wonder if you could line up the comments in this section?

File: README.md [hunk 2]
```diff
@@ -13,6 +13,7 @@ Options
 ----
 * **hash** - [Optional] A unicode string that will be used to generate the image. Defaults to a random hash based on the current time.
 * **options** - [Optional] An options object used to customize the generated image.
+    * **foreground** - The foreground color expressed as an rgba value array to use for the image foreground. For example, use [255,0,0,255] for red. Defaults to a color calculated from hash.
```
Comment (stewartlord) [inline README.md]: Can we reword this to make it more clear that the foreground is normally derived from the hash? Something like:
The foreground color is automatically derived from the hash value. Use this option to override that behavior and provide a rgba value array instead (e.g. [255,0,0,255] for red).

File: identicon.js [hunk 3]
```diff
@@ -53,11 +55,16 @@
...
             // light-grey background
-            var bg      = image.color(this.background[0], this.background[1], this.background[2], this.background[3]);
-
-            // foreground is last 7 chars as hue at 50% saturation, 70% brightness
-            var rgb     = this.hsl2rgb(parseInt(hash.substr(-7), 16) / 0xfffffff, 0.5, 0.7),
-                fg      = image.color(rgb[0] * 255, rgb[1] * 255, rgb[2] * 255);
+            var bg      = image.color(this.background[0], this.background[1], this.background[2], this.background[3]),
+                fg;
+
+            if(this.foreground) {
```
Comment (stewartlord) [inline identicon.js]: More nitpicking here. Can we add a space between if and the opening parenthesis to match style? Also if you could line up the assignments that would be lovely.

**Comment by stewartlord** [comment]
So sorry for the delay! New job has been keeping me busy. This looks great, just some minor tweaks suggested.

**Comment by redboul** [comment]
Hi !
Sorry, I was on vacation, thanks for your feedback. 
I just saw that I did not put any description on the PR, sorry about that. 
I needed to be able to set a custom color on the foreground so I added the feature if you're ok with it.

Thanks for your work on this project.

**Comment by gamazeps** [comment]
Hi !

Is there a plan for merging this PR ?
Indeed I kinda need that for a project and would rather use the package provided by npm rather than my own source.

Cheers

File: identicon.js [hunk 4]
```diff
@@ -34,12 +34,14 @@
 
         this.hash        = hash                    || this.defaults.hash;
         this.background  = this.options.background || this.defaults.background;
+        this.foreground  = this.options.foreground;
```
Comment (gamazeps) [inline identicon.js]: Add a default value for foreground.

**Comment by stewartlord** [comment]
Hey @gamazeps, sorry I didn't realize this had been updated. Thanks for the reminder!

**Comment by stewartlord** [comment]
@redboul thanks for the updates, just :ship: 'd this!

**Comment by gamazeps** [comment]
Awesome :)

**Comment by redboul** [comment]
Thx @stewartlord
